packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmAnyTests.kt

```diff
@@ -60,54 +69,320 @@ class RealmAnyTests {
-            val managedObj = copyToRealm(unmanagedObj)
-
-            assertEquals(unmanagedObj.stringField, managedObj.stringField)
-            assertEquals(unmanagedObj.byteField, managedObj.byteField)
-            assertEquals(unmanagedObj.charField, managedObj.charField)
-            assertEquals(unmanagedObj.shortField, managedObj.shortField)
-            assertEquals(unmanagedObj.intField, managedObj.intField)
-            assertEquals(unmanagedObj.longField, managedObj.longField)
-            assertEquals(unmanagedObj.booleanField, managedObj.booleanField)
-            assertEquals(unmanagedObj.floatField, managedObj.floatField)
-            assertEquals(unmanagedObj.doubleField, managedObj.doubleField)
-            assertEquals(unmanagedObj.timestampField, managedObj.timestampField)
-            assertEquals(unmanagedObj.objectIdField, managedObj.objectIdField)
-            assertEquals(unmanagedObj.realmObjectIdField, managedObj.realmObjectIdField)
-            assertEquals(unmanagedObj.uuidField, managedObj.uuidField)
-            assertContentEquals(unmanagedObj.byteArrayField, managedObj.byteArrayField)
-            assertEquals(unmanagedObj.objectField?.name, managedObj.objectField?.name)
-            assertEquals(
-                unmanagedObj.embeddedObjectField?.name,
-                managedObj.embeddedObjectField?.name
-            )
-            assertEquals(unmanagedObj.mutableRealmInt, managedObj.mutableRealmInt)
+            val unmanaged = IndexedRealmAnyContainer()
+            val managed = copyToRealm(unmanaged)
+            val realmAnyWithClassNotInSchema = RealmAny.create(notInSchema)
+            assertFailsWithMessage<IllegalArgumentException>("Schema does not contain a class named 'NotInSchema'") {
+                managed.anyField = realmAnyWithClassNotInSchema
+            }
         }
     }
-}
 
-class TestContainer : RealmObject {
-    var stringField: String? = "Realm"
-    var byteField: Byte? = 0xA
-    var charField: Char? = 'a'
-    var shortField: Short? = 17
-    var intField: Int? = 42
-    var longField: Long? = 256
-    var booleanField: Boolean? = true
-    var floatField: Float? = 3.14f
-    var doubleField: Double? = 1.19840122
-    var timestampField: RealmInstant? = RealmInstant.from(0, 0)
-    var objectIdField: BsonObjectId = BsonObjectId()
-    var realmObjectIdField: ObjectId? = ObjectId.create()
-    var uuidField: RealmUUID? = RealmUUID.random()
-    var byteArrayField: ByteArray? = byteArrayOf(42)
-    var objectField: TestParent? = TestParent()
-    var embeddedObjectField: TestEmbeddedChild? = TestEmbeddedChild()
-    var mutableRealmInt: MutableRealmInt? = MutableRealmInt.create(42)
+    @Test
+    fun unmanaged_incorrectTypeThrows() {
+        val realmAny = RealmAny.create(10.toShort())
+        assertFails {
+            realmAny.asString()
+        }.let {
+            val kajshd = 0
+        }
+    }
+
+    @Test
+    fun unmanaged_short() {
```

Comment:
Is there a way for us to iterate this making sure we update them if needed, e.g. by using the `RealmAny.Type` enum?